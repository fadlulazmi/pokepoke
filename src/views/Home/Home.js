import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getList } from '../../store/action';
import Card from '../../components/Card';
import './styles.scoped.css';
import { Link } from 'react-router-dom';

export default function Home() {
  const dispatch = useDispatch();
  const state = useSelector(state => state);
  const {listPokemon, nextFetchedList} = state
  useEffect(() => {
    dispatch(getList());
  }, []);
  const getNewList = () => dispatch(getList(state.nextFetchedList));
  console.log(state);
  return (
    <section className="root">
      <div className="container">
        { listPokemon.map(el => (
          <Link to={`/detail/${el.name}`}>
            <Card
              key={el.name}
              name={el.name}
              counter={el.owned}
            />
          </Link>
          ))
        }
      </div>
      <br/>
      <button onClick={() => getNewList()}>next</button>
    </section>
  );
}
