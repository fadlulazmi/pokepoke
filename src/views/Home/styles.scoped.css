.root {
  display: flex;
  flex-direction: column;
  align-content: center;
  align-items: center;
}

.container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  padding: 10px;
  justify-content: space-evenly;
}

.root > .button {
  border: none;
  background-color: none;
  height: 30px;
  width: 300px;
  margin-bottom: 100px;
}