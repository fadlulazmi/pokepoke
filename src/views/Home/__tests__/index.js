import Home from '../Home'
import ShallowRenderer from 'react-test-renderer/shallow';
import { useSelector } from 'react-redux';


describe('src/components/Home', () => {
  test('should be rendered correctly', () => {
    useSelector.mockImplementationOnce(fn => {
      fn()
      return {
        listPokemon: [],
        nextFetcehdList: ''
      }
    })
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Home />);
    expect(tree).toMatchSnapshot();
  })
})
