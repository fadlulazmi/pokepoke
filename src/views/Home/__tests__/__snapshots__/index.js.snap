// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Home should be rendered correctly 1`] = `
<section
  className="root"
>
  <div
    className="container"
  />
  <br />
  <p>
    You reached the end of page
  </p>
</section>
`;
